framework/scene_graph/components/image.h

```diff
@@ -146,6 +144,8 @@ class Image : public Component
 	std::unique_ptr<core::Image> vk_image;
 
 	std::unique_ptr<core::ImageView> vk_image_view;
+
+	size_t data_hash{0};
```

Comment:
Sorry, but this doesn't fix the hpp vs. non-hpp issue.
The data layout of those two classes need to be identical. That is, you need to add the data_hash at the same position to both, the Image and the HPPImage class.

Reply:
Yes that was it, fixed, thank you!